ddd-koans/Value Object/Static Factory & Reflect Domain Representation/src/Main.kt

```diff
@@ -46,4 +23,16 @@ class ValidTimeSignature(private val numerator: Int, private val denominator: In
     }
 
     override fun toString(): String = "TimeSignature($numerator/$denominator)"
-}
+
+    companion object {
+        fun of(timeSignature: String): TimeSignature {
+            val timeSignatureRegex = "^(\\d{1,2})/(\\d{1,2})$".toRegex()
+            val groupValues = timeSignatureRegex.matchEntire(timeSignature)?.groupValues
+            return groupValues?.let {
+                val (numerator, denominator) = (groupValues[1] to groupValues[2])
+                TimeSignature(numerator.toInt(), denominator.toInt())
+            } ?: throw IllegalArgumentException("The numerator and denominator must be present")
```

Comment:
can any exception be thrown here?

Reply:
Yes, but I guess we should create our own exception?

Reply:
probably yes

Reply:
InvalidTimeSignatureException ?